hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+        headers = {"Loopring-Authentication": auth}
+        return headers
+    
+
+    async def _http_client(self) -> aiohttp.ClientSession:
+        if self._shared_client is None:
+            self._shared_client = aiohttp.ClientSession()
+        return self._shared_client
+    
+
+    async def _api_request(self,
+                           http_method: str,
+                           url: str,
+                           data: Optional[Dict[str, Any]] = None,
+                           params: Optional[Dict[str, Any]] = None,
+                           headers: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
+        client = await self._http_client()
+        async with client.request(http_method, url=url, timeout=self.API_CALL_TIMEOUT, data=data, params=params,
+                                  headers=headers) as response:
+            if response.status != 200:
+                raise IOError(f"Error fetching data from {url}. HTTP status is {response.status}.")
+            data = await response.json()
+
+            # Keep an auto-expired record of the response and the request URL for debugging and logging purpose.
+            self._api_response_records[url] = response
+
+            return data
+        
+        
+
+    async def _update_trade_fees(self):
+        cdef:
+            double current_timestamp = self._current_timestamp
+
+        if current_timestamp - self._last_update_trade_fees_timestamp > self.UPDATE_TRADE_FEES_INTERVAL or self._gas_fee_usd == NaN:
+            
+            
+            calc_fee_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/info"
+            
+            
+            res = await self._api_request(http_method="get", url=calc_fee_url)
+            
+            self._maker_trade_fee = float(res["data"]["maker_fee_percentage"])
+            self._taker_trade_fee = float(res["data"]["taker_fee_percentage"])
+            
+            
+            self._gas_fee_weth = calculateGasFees("WETH")
+            self._gas_fee_usd = calculateGasFees("DAI")
+            
+            
+            self._last_update_trade_fees_timestamp = current_timestamp
+            
+
+    cdef object c_get_fee(self,
+                          str base_currency,
+                          str quote_currency,
+                          object order_type,
+                          object order_side,
+                          double amount,
+                          double price):
+        cdef:
+            double gas_fee = 0.0 #Upper bound? 
+            double percent
+            
+        if quote_currency == "WETH":
+            gas_fee = self._gas_fee_weth
+        elif quote_currency == "DAI":
+            gas_fee = self._gas_fee_usd
+        else:
+            self.logger().warning(
+                f"Unrecognized quote token symbol - {quote_currency}. Assuming gas fee is in stable coin units."
+            )
+            gas_fee = NaN 
```

Comment:
Is this necessary? In the future, fees can be any token in a market

Reply:
Okay that warning was just part of the original Hummingbot code in case of an invalid parameter but we can make sure to add the additional tokens in the future